engine/item_effects.asm

```diff
@@ -314,6 +314,9 @@ KeyItemEffects:
 	dw ApricornBox      ; APRICORN_BOX
 
 PokeBallEffect: ; e8a2
+	ld a, [wBattleMode]
+	cp 0 ; overworld
```

Comment:
`and a` is the standard [optimized](https://github.com/pret/pokecrystal/wiki/Optimizing-assembly-code#compare-a-to-0) way to test for 0 instead of `cp 0`.